### PR TITLE
Refactor for CAS-500

### PR DESCRIPTION
**Pull request by blaisedias**
Use and move functions from and to common/util.go
Add pvc_stress test that uses fio

### CODE REVIEW

File: mayastor-test/e2e/nightly/pvc_stress_fio/pvc_stress_fio_test.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+//	3. The associated MV is created and has a State "healthy"
+//  4. That a test application (fio) can read and write to the volume
+// then Delete the PVC and verify that
+//	1. The PVC is deleted
+//	2. The associated PV is deleted
+//  3. The associated MV is deleted
+func testPVC(volName string, scName string) {
+	fmt.Printf("%s, %s\n", volName, scName)
+	// PVC create options
+	createOpts := &coreV1.PersistentVolumeClaim{
```
Comment (chriswldenyer) [inline mayastor-test/e2e/nightly/pvc_stress_fio/pvc_stress_fio_test.go:35]: You can use Cmn,MkPVC() and Cmn.RmPVC() for a lot of this code.

**Review by chriswldenyer — APPROVED** [review]
LGTM, some scope for more common code.

File: mayastor-test/e2e/common/util.go [hunk 2]
```diff
@@ -352,3 +352,80 @@ func FioReadyPod() bool {
...
+func IsPodRunning(podName string) bool {
+	var pod corev1.Pod
+	if gTestEnv.K8sClient.Get(context.TODO(), types.NamespacedName{Name: podName, Namespace: "default"}, &pod) != nil {
+		return false
+	}
+	return pod.Status.Phase == v1.PodRunning
+}
+
+/// Create a PVC in default namespace, no options and no context
+func CreatePVC(pvcIn *v1.PersistentVolumeClaim) (*v1.PersistentVolumeClaim, error) {
```
Comment (paulyoong) [inline mayastor-test/e2e/common/util.go:365]: Nit - why "pvcIn" and not just "pvc"?

File: mayastor-test/e2e/common/util.go [hunk 3]
```diff
@@ -352,3 +352,80 @@ func FioReadyPod() bool {
...
+func CreatePod(podDef *corev1.Pod) (*corev1.Pod, error) {
+	return gTestEnv.KubeInt.CoreV1().Pods("default").Create(context.TODO(), podDef, metav1.CreateOptions{})
+}
+
+/// Delete a Pod in default namespace, no options and no context
+func DeletePod(podName string) error {
+	return gTestEnv.KubeInt.CoreV1().Pods("default").Delete(context.TODO(), podName, metav1.DeleteOptions{})
+}
+
+/// Create a test fiod pod in default namespace, no options and no context
```
Comment (paulyoong) [inline mayastor-test/e2e/common/util.go:394]: typo "fiod"

File: mayastor-test/e2e/common/util.go [hunk 4]
```diff
@@ -45,13 +45,13 @@ func LabelNode(nodename string, label string) {
 }
 
 // Status part of the mayastor volume CRD
-type mayastorVolStatus struct {
-	state  string
+type MayastorVolStatus struct {
+	State  string
```
Comment (paulyoong) [inline mayastor-test/e2e/common/util.go:49]: Nit - what is the reason for capitilising "State" but not the other members? It just looks a little inconsistent.

File: mayastor-test/e2e/nightly/pvc_stress_fio/pvc_stress_fio_test.go [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+	).Should(Equal(true))
+}
+
+func stressTestPVC() {
+	for ix := 0; ix < 10; ix++ {
+		testPVC(fmt.Sprintf("stress-pvc-nvmf-%d", ix), "mayastor-nvmf")
+		testPVC(fmt.Sprintf("stress-pvc-iscsi-%d", ix), "mayastor-iscsi")
+		// FIXME: Without this delay GetPvcClaimStatusPhase returns Pending
+		// even though kubectl shows that the pvc is Bound.
+		//pause()
```
Comment (paulyoong) [inline mayastor-test/e2e/nightly/pvc_stress_fio/pvc_stress_fio_test.go:166]: Can this be removed as the delay isn't used?

**Review by paulyoong — APPROVED** [review]

File: mayastor-test/e2e/common/util.go (hunk 2 above)
Comment (blaisedias) [reply mayastor-test/e2e/common/util.go]: fixed

File: mayastor-test/e2e/common/util.go (hunk 4 above)
Comment (blaisedias) [reply mayastor-test/e2e/common/util.go:49]: Fixed, all capitalised for export reasons.

File: mayastor-test/e2e/nightly/pvc_stress_fio/pvc_stress_fio_test.go (hunk 5 above)
Comment (blaisedias) [reply mayastor-test/e2e/nightly/pvc_stress_fio/pvc_stress_fio_test.go]: Done

**Comment by blaisedias** [comment]
bors merge

**Pull request merged**